app/src/main/java/com/example/encoratask/ItemListActivity.kt

```diff
@@ -43,23 +56,27 @@ class ItemListActivity : AppCompatActivity() {
                     .setAction("Action", null).show()
         }
 
-        setupRecyclerView(findViewById(R.id.item_list))
+
+        viewModel.listCharacter.observe(this, Observer {
+            setupRecyclerView(findViewById(R.id.item_list), it)
+        })
     }
 
-    private fun setupRecyclerView(recyclerView: RecyclerView) {
-        recyclerView.adapter = SimpleItemRecyclerViewAdapter(DummyContent.ITEMS)
+    private fun setupRecyclerView(recyclerView: RecyclerView, list: List<Character>) {
+        recyclerView.adapter = SimpleItemRecyclerViewAdapter(list)
     }
 
-    class SimpleItemRecyclerViewAdapter(private val values: List<DummyContent.DummyItem>) :
+    class SimpleItemRecyclerViewAdapter(private val values: List<Character>) :
             RecyclerView.Adapter<SimpleItemRecyclerViewAdapter.ViewHolder>() {
 
         private val onClickListener: View.OnClickListener
 
         init {
             onClickListener = View.OnClickListener { v ->
-                val item = v.tag as DummyContent.DummyItem
+                val item = v.tag as Character
                 val intent = Intent(v.context, ItemDetailActivity::class.java).apply {
-                    putExtra(ItemDetailFragment.ARG_ITEM_ID, item.id)
+                    putExtra("NAME", item.name)
```

Comment:
Also, If you're thinking on passing multiple values of the same nature (in this case, are attributes of a character) I suggest to pass the whole object and make the data class parcelable